compendium/DeclarativeServices/test/TestComponentLifecycle.cpp

```diff
@@ -175,13 +187,65 @@ TEST_F(tServiceComponent, testImmediateComponent_LifeCycle_Dynamic) // DS_TOI_51
   EXPECT_FALSE(static_cast<bool>(sRef2)) << "Service must not be available after it's dependency is removed";
 }
 
+//Function to validate lazy loading of delayed component
+std::size_t lazyLoadingValidation()
+{
+  #if defined(_WIN32)
+
+    HMODULE hMods[1024];
+    DWORD cbNeeded;
+    unsigned int i;
+
+    HANDLE hProcess = GetCurrentProcess();
+    auto res = EnumProcessModules(hProcess, hMods, sizeof(hMods), &cbNeeded);
+    EXPECT_GT(sizeof(hMods), cbNeeded) << "Size of array is too small to hold all module handles";
```

Comment:
`EnumProcessModules` returns the number of bytes required to store all module handles in `cbNeeded`. There shouldn't be a need to test for the size. Instead if the size is too small, call `EnumProcessModules` with a sufficiently large enough buffer.